### PR TITLE
Google docstring conversion batch 8

### PR DESCRIPTION
**Pull request by michael-berk**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/michael-berk/mlflow/pull/11042?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11042/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11042
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Conversion for all remaining directories in `mlflow` (other than mleap). Mleap was not included because the deprecation formatter breaks everything (including the current production formatting).

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/docstring_utils.py [hunk 1]
```diff
@@ -127,30 +125,15 @@ def format_docstring(self, docstring: str) -> str:
...
+    Returns a decorator that replaces param doc placeholders (e.g. '{{ param_name }}') in the
     docstring of the decorated function.
 
     Args:
         param_docs: A `ParamDocs` instance or `dict`.
 
     Returns:
         A decorator to apply the formatting.
-
-    Examples:
```
Comment (michael-berk) [inline mlflow/utils/docstring_utils.py:139]: Resolved

File: mlflow/server/handlers.py [hunk 2]
```diff
@@ -357,22 +364,23 @@ def _assert_item_type_string(x):
...
+        param: The string name of the parameter being validated.
+        value: The corresponding value of the `param` being validated.
+        proto_parsing_succeeded: A boolean value indicating whether proto parsing succeeded.
+            If the proto was successfully parsed, we assume all of the types of the parameters in
+            the request body were correctly specified, and thus we skip validating types. If proto
+            parsing failed, then we validate types in addition to the rest of the schema. For
+            details, see https://github.com/mlflow/mlflow/pull/5458#issuecomment-1080880870.
+
+    Returns:
+        None on validation success. Otherwise, raises an MLFlowException if an assertion fails.
```
Comment (michael-berk) [inline mlflow/server/handlers.py:383]: Resolved

File: mlflow/server/handlers.py [hunk 3]
```diff
@@ -2108,8 +2117,11 @@ def _get_paths(base_path):
 
 def get_handler(request_class):
     """
-    :param request_class: The type of protobuf message
-    :return:
+    Args:
+        request_class: The type of protobuf message
+
+    Returns:
```
Comment (michael-berk) [inline mlflow/server/handlers.py:2123]: Resolved

File: mlflow/server/auth/client.py [hunk 4]
```diff
@@ -414,39 +356,14 @@ def get_registered_model_permission(self, name: str, username: str):
...
-        :param username: The username.
-
-        :return: A single :py:class:`mlflow.server.auth.entities.RegisteredModelPermission` object.
-                 Raises ``RestException`` if the user does not exist,
-                 or no permission exists for this registered model user pair.
-                 Note that the default permission will still be effective even if
-                 no permission exists.
-
-        .. code-block:: bash
-            :caption: Example
```
Comment (michael-berk) [inline mlflow/server/auth/client.py:427]: Resolved

File: mlflow/server/auth/client.py [hunk 5]
```diff
@@ -209,40 +193,17 @@ def create_experiment_permission(self, experiment_id: str, username: str, permis
...
-        :param permission: Permission to grant.
-            Must be one of "READ", "EDIT", "MANAGE" and "NO_PERMISSIONS".
-
-        :return: A single :py:class:`mlflow.server.auth.entities.ExperimentPermission` object.
-                 Raises ``RestException`` if the user does not exist,
-                 or a permission already exists for this experiment user pair,
-                 or if the permission is invalid.
-                 Does not require ``experiment_id`` to be an existing experiment.
-
-        .. code-block:: bash
```
Comment (michael-berk) [inline mlflow/server/auth/client.py:223]: Resolved

File: mlflow/server/auth/client.py [hunk 6]
```diff
@@ -77,37 +80,12 @@ def get_user(self, username: str):
...
         Get a user with a specific username.
 
-        :param username: The username.
+        Args:
+            username: The username.
 
-        :return: A single :py:class:`mlflow.server.auth.entities.User` object.
-                 Raises ``RestException`` if the user does not exist.
-
-        .. code-block:: bash
```
Comment (michael-berk) [inline mlflow/server/auth/client.py:85]: Resolved

File: mlflow/server/auth/__init__.py [hunk 7]
```diff
@@ -655,11 +657,11 @@ def alert(href: str):
 {% with messages = get_flashed_messages() %}
   {% if messages %}
     {% for message in messages %}
-      alert("{{ message }}");
```
Comment (michael-berk) [inline mlflow/server/auth/__init__.py:658]: Resolved

File: mlflow/pyfunc/scoring_server/client.py [hunk 8]
```diff
@@ -73,12 +77,15 @@ def wait_server_ready(self, timeout=30, scoring_server_proc=None):
...
     def invoke(self, data, params: Optional[Dict[str, Any]] = None):
         """
-        :param data: Model input data.
-        :param params: Additional parameters to pass to the model for inference.
+        Args:
+            data: Model input data.
+            params: Additional parameters to pass to the model for inference.
 
-                       .. Note:: Experimental: This parameter may change or be removed in a future
-                                               release without warning.
```
Comment (michael-berk) [inline mlflow/pyfunc/scoring_server/client.py:80]: Resolved

File: mlflow/pyfunc/model.py [hunk 9]
```diff
@@ -49,19 +49,20 @@
 
 def get_default_pip_requirements():
     """
-    :return: A list of default pip requirements for MLflow Models produced by this flavor.
```
Comment (michael-berk) [inline mlflow/pyfunc/model.py:52]: Resolved

File: mlflow/pyfunc/__init__.py [hunk 10]
```diff
@@ -518,46 +522,8 @@ def unwrap_python_model(self):
         This method is useful for accessing custom model functions, while still being able to
         leverage the MLflow designed workflow through the `predict()` method.
 
-        :return: The underlying wrapped model object
-
-        .. testcode:: python
```
Comment (michael-berk) [inline mlflow/pyfunc/__init__.py:523]: Resolved

File: mlflow/projects/__init__.py [hunk 11]
```diff
@@ -214,97 +217,51 @@ def run(
     """
     Run an MLflow project. The project can be local or stored at a Git URI.
 
-    MLflow provides built-in support for running projects locally or remotely on a Databricks or
```
Comment (michael-berk) [inline mlflow/projects/__init__.py:217]: Resolved

File: mlflow/models/model.py [hunk 12]
```diff
@@ -642,44 +647,24 @@ def get_model_info(model_uri: str) -> ModelInfo:
...
+            - ``models:/<model_name>/<stage>``
+            - ``mlflow-artifacts:/path/to/model``
 
-    :return: A :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
-            metadata of the logged model.
+            For more information about supported URI schemes, see
+            `Referencing Artifacts <https://www.mlflow.org/docs/latest/concepts.html#
+            artifact-locations>`_.
 
-    .. code-block:: python
```
Comment (michael-berk) [inline mlflow/models/model.py:662]: Resolved

File: mlflow/metrics/genai/genai_metric.py [hunk 13]
```diff
@@ -97,94 +97,36 @@ def make_genai_metric(
...
-        setting the temperature to 0.0 for the LLM used as a judge to ensure consistent results.
-    :param aggregations: (Optional) The list of options to aggregate the scores. Currently supported
-        options are: min, max, mean, median, variance, p90.
-    :param greater_is_better: (Optional) Whether the metric is better when it is greater.
-    :param max_workers: (Optional) The maximum number of workers to use for judge scoring.
-        Defaults to 10 workers.
-
-    :return: A metric object.
-
-    .. testcode:: python
```
Comment (michael-berk) [inline mlflow/metrics/genai/genai_metric.py:128]: Resolved

**Comment by michael-berk** [comment]
Manual diff looks good.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**